ops/testing.py

```diff
@@ -1634,6 +1634,9 @@ def evaluate_status(self) -> None:
         Tests should normally call this and then assert that ``self.model.app.status``
         or ``self.model.unit.status`` is the value expected.
         """
+        if self._backend._is_leader:
+            self.charm.app._collected_statuses = []
```

Comment:
The distinction for leader seems inconsistent with the `clear`:
If `evaluate_status` clears the stage for the next `add_status`, it's a harness-level function. Leadership status is on a different level.

Reply:
Yeah, good call. I've now cleared this independent of `is_leader`.